File: project10/java/Penguin.java
/* Penguin.java provides an Penguin subclass of Walkingbird.
 *
 * Completed by: Haim Hong
 * Date: March 25 2022
 ******************************************************/

public class Penguin extends WalkingBird{
    /* default constructor
     * PostCond: myName == "".
     */
     public Penguin(){
       super();
     }
    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
     public Penguin(String name){
       super(name);
     }

   
   
    /* An Penguin's Call
     * Return: an owl-call ("Whoo-hoo!").
     */
     public String call(){
       return "Huh-huh-huh-huuuuh!";
     }
   } 

File: project10/ruby/Penguin.rb
# Penguin.rb | Defines the Penguin class which inherits attributes and methods
#
# Completed by: Haim Hong
# Date: March 25 2022
####################################################

require './Walkingbird.rb'

class Penguin < Walkingbird
    def call # call function for this class specific
        'Snork!'
    end
end